### PR TITLE
1297047: Candlepin allows import of expired manifest

### PR DESCRIPTION
**Pull request by vritant**
* Return an ImportRecord upon import success.
* The status of the record marks if:
  * There were no active subscriptions in the manifest
  * There were one or more inactive subscriptions in the manifest

### CODE REVIEW

**Comment by wottop** [comment]
The bug says it should fail if expired subs. I would at least expect it to fail if all are expired.
If this is not the case based on discussions, please update the bug.

**Comment by vritant** [comment]
@wottop done.

File: server/src/main/java/org/candlepin/audit/Eventful.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+ * Marker Interface for objects which may produce events
+ */
+public interface Eventful {
+
+    /**
+     * Gets the id of an object
+     * @return The id of the object
+     */
+    Object getId();
+}
```
Comment (wottop) [inline server/src/main/java/org/candlepin/audit/Eventful.java]: Why introduce another interface here? It is using the information you already have in AbstractHibernateObject. Semantics?

Comment (vritant) [inline server/src/main/java/org/candlepin/audit/Eventful.java]: @wottop to let us send out events for DTOs

File: server/spec/import_warning_spec.rb [hunk 2]
```diff
@@ -0,0 +1,87 @@
+require 'spec_helper'
+require 'candlepin_scenarios'
+require 'json'
+require 'zip/zip'
+
+describe 'Import', :serial => true do
```
Comment (wottop) [inline server/spec/import_warning_spec.rb]: This needs to be 'Import Warning'

**Comment by wottop** [comment]
Added manifest with expired entitlement to the bug. It runs correctly. Fix the note above and I'll ack it.

**Comment by wottop** [comment]
I attached a manifest to the bug that has only an expired cert. I imported it and it behaved as expected.

It also wiped out all the test data pools. If we import a fully expired manifest, do we want to leave the existing pools alone?

**Comment by vritant** [comment]
@wottop I assumed that was the expected behavior as we have an [existing spec test](https://github.com/candlepin/candlepin/blob/master/server/spec/import_update_spec.rb#L45) to ensure that behavior.

**Comment by wottop** [comment]
I'll accept that. it does seem like a consequence that a user will complain about. It will wipe out all subs and entitlements for no net gain. I guess they can do that now if they import the wrong file.

You still need to get the describe tag in import_warning_spec.

File: server/spec/import_warning_spec.rb (hunk 2 above)
Comment (vritant) [inline server/spec/import_warning_spec.rb]: @wottop fixed, sorry for the delay!

**Comment by wottop** [comment]
ack

**Pull request merged**